onsite/lucxor/flr.py

```diff
@@ -1,798 +1,798 @@
+        self.delta_score_to_flr_map = {}  # delta_score -> (global_flr, local_flr)
+        
+        logger.debug("FLRCalculator initialized")
+    
+    def add_psm(self, delta_score: float, is_decoy: bool) -> None:
+        """
+        Add PSM to calculator
+        
+        Args:
+            delta_score: Delta score
+            is_decoy: Whether it's a decoy sequence
+        """
+        logger.debug(f"Adding PSM - delta_score: {delta_score}, is_decoy: {is_decoy}")
+        
+        if delta_score > self.max_delta_score:
+            self.max_delta_score = delta_score
+            
+        if is_decoy:
+            self.decoy_psms.append(delta_score)
+            self.n_decoy += 1
+        else:
+            self.real_psms.append(delta_score)
+            self.n_real += 1
+            
+        logger.debug(f"Current real_psms count: {self.n_real}, decoy_psms count: {self.n_decoy}")
+    
+    def prep_arrays(self) -> None:
+        """Prepare arrays"""
+        self.pos = np.array(self.real_psms)
+        self.neg = np.array(self.decoy_psms)
+        self.n_real = len(self.pos)
+        self.n_decoy = len(self.neg)
+    
+    def initialize_tick_marks(self) -> None:
+        """Initialize tick marks"""
+        # First prepare arrays
+        self.prep_arrays()
+        
+        self.max_delta_score *= 1.001  # Need to be slightly larger for binning
+        
+        self.tick_marks = np.zeros(self.NMARKS)
+        for i in range(self.NMARKS):
+            x = (i * self.max_delta_score) / self.NMARKS
+            self.tick_marks[i] = x
+        
+        # Initialize other arrays
+        self.local_fdr = np.zeros(self.n_real)
+        self.global_fdr = np.zeros(self.n_real)
+        
+        self.calc_delta_score_mean()
+        self.calc_delta_score_var()
+        
+        self.get_bandwidth(DECOY)  # Decoy
+        self.get_bandwidth(REAL)   # Target
+        
+        # Output bandwidth information
+        logger.info(f"FLR bandwidth (pos): {self.bw_real:.6f}")
+        logger.info(f"FLR bandwidth (neg): {self.bw_decoy:.6f}")
+        
+
+    
+    def get_bandwidth(self, data_type: int) -> None:
+        """
+        Calculate bandwidth
+        
+        Args:
+            data_type: Data type (REAL or DECOY)
+        """
+        if data_type == REAL:
+            sigma = np.sqrt(self.delta_score_var_pos)
+            N = float(self.n_real)
+            x = np.power(N, 0.2)
+            result = 1.06 * (sigma / x)
+            self.bw_real = result
+            
+        elif data_type == DECOY:
+            sigma = np.sqrt(self.delta_score_var_neg)
+            N = float(self.n_decoy)
+            x = np.power(N, 0.2)
+            result = 1.06 * (sigma / x)
+            self.bw_decoy = result
+    
+    def calc_delta_score_mean(self) -> None:
+        """Calculate delta score mean"""
+        # Target sequence
+        sum_pos = 0.0
+        for d in self.pos:

```

Comment:
np.sum?

Reply:
Do you really need to store N or just use len in the rest of the code and np.mean here